test/test_helpers.py

```diff
@@ -74,6 +77,95 @@ def fake_script_calls(test_case: unittest.TestCase,
     return calls  # type: ignore
 
 
+def create_framework(
+        monkeypatch: pytest.MonkeyPatch,
+        request: pytest.FixtureRequest,
+        *,
+        meta: typing.Union[ops.CharmMeta, None] = None):
+    monkeypatch.setenv("PATH", str(pathlib.Path(__file__).parent / 'bin'), prepend=os.pathsep)
+    monkeypatch.setenv("JUJU_UNIT_NAME", "local/0")
+
+    tmpdir = pathlib.Path(tempfile.mkdtemp())
+    request.addfinalizer(functools.partial(shutil.rmtree, str(tmpdir)))
```

Comment:
I think if rmtree throws an error, it will be passed to the test, which is captured and reported as a test failure, right? Same as unittest addCleanup? Do we need to surrounded it with a try/except?

Reply:
Resolving this because if rmtree fails, the test case will too. Confirmed with a simple test case:

```python
import pytest


def test_with_fixture(request: pytest.FixtureRequest):
    def finalizer():
        raise Exception()

    request.addfinalizer(finalizer)

    assert True

```